src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java

```diff
@@ -0,0 +1,128 @@
+import uk.gov.hmcts.reform.civil.config.StitchingConfiguration;
+import uk.gov.hmcts.reform.civil.enums.YesOrNo;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.*;
+import uk.gov.hmcts.reform.civil.model.documents.CaseDocument;
+import uk.gov.hmcts.reform.civil.model.documents.Document;
+import uk.gov.hmcts.reform.civil.model.documents.DocumentMetaData;
+import uk.gov.hmcts.reform.civil.sampledata.CaseDataBuilder;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphabetic;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.civil.model.documents.DocumentType.SEALED_CLAIM;
+import static uk.gov.hmcts.reform.civil.service.docmosis.DocmosisTemplates.N1;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {
+    CivilDocumentStitchingService.class,
+    JacksonAutoConfiguration.class,
+    CaseDetailsConverter.class
+})
+public class CivilDocumentStitchingServiceTest {
+    @Autowired
+    private CivilDocumentStitchingService civilDocumentStitchingService;
+    @MockBean
+    private BundleRequestExecutor bundleRequestExecutor;
+    @MockBean
+    private StitchingConfiguration stitchingConfiguration;
+    private static final CaseDocument CLAIM_FORM =
+        CaseDocument.builder()
+            .createdBy("John")
+            .documentName(String.format(N1.getDocumentTitle(), "000DC001"))
+            .documentSize(0L)
+            .documentType(SEALED_CLAIM)
+            .createdDatetime(LocalDateTime.now())
+            .documentLink(Document.builder()
+                              .documentUrl("fake-url")
+                              .documentFileName("file-name")
+                              .documentBinaryUrl("binary-url")
+                              .build())
+            .build();
+    //Optional should be removed after refactoring
+    private final Optional<String> stitchStatus = Optional.of(randomAlphabetic(8));
+    private final Optional<Document> stitchedDocument = Optional.of(mock(Document.class));
+
+    private List<DocumentMetaData> buildDocumentMetaDataList() {
+        return List.of(
+            new DocumentMetaData(
+                CLAIM_FORM.getDocumentLink(),
+                "Sealed Claim Form",
+                LocalDate.now().toString()
+            )
+        );
+    }
+
+    public CaseData buildCaseData(Bundle bundle){
+        List<IdValue<Bundle>> caseBundles = List.of(new IdValue<>("1", bundle));
+        return CaseDataBuilder.builder().caseBundles(caseBundles).build();
+    }
+    @Test
+    public void whenCaseDataIsNullWithPayloadAndNoCaseBundlesNoStitchingDocument() {
```

Comment:
The `public` modifier is not necessary in JUnit Jupiter tests. Remove it from all the tests.

Reply:
Done